update notebook to use new search engine

jburel opened the pull request:
Use New search engine

pwalczysko commented:
The notebook works exactly as it was working before, tested on MyBinder.and repo2docker.

Things to improve in a follow-up PR:
1. Add note about conda install (the repo2docker only is suggested both in README of the repo as well as again, inside the notebook itself)
2. Use the result found of https://idr-testing.openmicroscopy.org/webclient/?show=image-4386966 from HPA, point out that the "Pathology" is "Normal tissue" - find any other images of PDX1 from Pancreas or Islets with "Pathology: SOME DISEASE" and make the comparison in the notebook
3. Search for "Pancreas", but leave the search for "Islets" in as well, so as to show that sometimes you have to be inventive when creating your search query

jburel commented:
https://github.com/IDR/idr-notebooks/issues/117

pwalczysko commented on humanmine.ipynb:
```diff
@@ -331,206 +341,390 @@
+    "        for v in values:\n",
+    "            name = v[\"name\"]\n",
+    "            value = v['value']\n",
+    "            if PART in name and (TISSUE or EXPRESSION in value):\n",
```
This line throws an error

```
NameError                                 Traceback (most recent call last)
/tmp/ipykernel_61/654637249.py in <module>
      9             name = v["name"]
     10             value = v['value']
---> 11             if PART in name and (TISSUE or EXPRESSION in value):
     12                 part = value
     13         for v in values:

NameError: name 'PART' is not defined
```

pwalczysko replied:
(I tried to fix it by de-capitalizing the undefined PART variable, but this then threw error complaining about the ``part`` being None.

pwalczysko replied:
Aha, sorry, got it, it should be ``PART_KEY`` as defined in the top of the notebook.
Will make a suggestion.

pwalczysko commented on humanmine.ipynb:
```diff
@@ -331,206 +341,390 @@
-    "## Find the images\n",
-    "Find the images linked to the selected genes in one the project found.\n",
-    "Below, we look at the gene **PDX1**"
+    "**Now we filter images with a given Organism Part value**.\n",
```
I just cannot find any place where the ``Organism Part`` value is being set. How does the value from ``TISSUE = "Pancreas" # "Cerebellum" # "little brain"`` which is used in Humanmine pass into the variable ``PART_KEY`` is not clear to me.

pwalczysko replied:
Ah, sorry, ignore, I can see it in line ``if PART_KEY in name and (TISSUE or EXPRESSION in value)``

pwalczysko commented on humanmine.ipynb:
```diff
@@ -331,206 +341,390 @@
+    "    'Disease: ', True\n",
+    ")\n",
+    "\n",
+    "button_next = widgets.Button(description=\"Next>>\")\n",
```
The Next and Previous buttons have a bit erratic behavour. The number of images before and after a sequence of clicks ``Next`` -> ``Previous`` does not necessarily match. Sometimes the clicking of Next or Previous get the widget into a bad state, which can be gotten rid of only by selecting another disease from the dropdown.

pwalczysko commented on humanmine.ipynb:
```diff
@@ -331,206 +341,390 @@
+    "        max_value = increase \n",
+    "        display_images(values, 0, max_value)\n",
+    "                \n",
+    "def get_images(disease):\n",
```
Could we please have the number of images available form each disease displayed immediately once I select the disease in the dropdown ? This would improve the experience and give the user a hand when he/she gets lost due to the "Next" and "Previous" buttons weirdness, see https://github.com/IDR/idr-notebooks/pull/116/files#r903566397

pwalczysko commented on humanmine.ipynb:
```diff
@@ -331,206 +341,390 @@
+    "        if k == disease:\n",
+    "            return v\n",
+    "    return None\n",
+    "    \n",
```
Above or below the images, a simple line informing the users that the images are clickable and the click leads to the image in IDR is necessary here imho, otherwise the user might not click.

pwalczysko commented:
Looks great, just made couple of comments to improve the user experience and understanding.

pwalczysko commented on humanmine.ipynb:
```diff
@@ -331,206 +329,412 @@
+    "import matplotlib.pyplot as plt\n",
+    "disease_list = []\n",
+    "disease_list_values = []\n",
+    "gene = \"PDX1\"\n",
```
When the gene here is replaced with a gene where no images are found, this cell crashes. Maybe catch and say "no images found for this gene" instead ?

pwalczysko commented:
![Screenshot 2022-06-23 at 11 25 50](https://user-images.githubusercontent.com/2478303/175280095-239b720b-ade5-4959-b0b2-9336c48e749f.png)

The graph makes sense, but the X axis description is too hard to read. Could we have a more spaced columns and some slanting of the description, so that one does not have to twist the neck so hard ?

pwalczysko commented:
![Screenshot 2022-06-23 at 11 39 42](https://user-images.githubusercontent.com/2478303/175280522-a3cc8def-aa07-40d6-9c93-7c2d0de85b38.png)
Minor comment on this cell - could the list above be alphabetically sorted ?

But otherwise this output is great imho, it really gives the user the overview and the Next and Previous buttons work nicely too.

pwalczysko commented:
Three more comments, https://github.com/IDR/idr-notebooks/pull/116#discussion_r904864977 https://github.com/IDR/idr-notebooks/pull/116#issuecomment-1164250848,   https://github.com/IDR/idr-notebooks/pull/116#issuecomment-1164253082

jburel commented:
> ![Screenshot 2022-06-23 at 11 25 50](https://user-images.githubusercontent.com/2478303/175280095-239b720b-ade5-4959-b0b2-9336c48e749f.png)
> 
> The graph makes sense, but the X axis description is too hard to read. Could we have a more spaced columns and some slanting of the description, so that one does not have to twist the neck so hard ?

I will have to see what I can do but I do not want to introduce too much custom code to achieve that. I use the default ``plot`` method from ``pandas.DataFrame``

jburel commented:
@pwalczysko I found a way to rotate the plot so it is easier to read and add space

pwalczysko commented:
Thanks, reads much better.

jburel commented:
Thanks @pwalczysko for the review. I will change the url when we are ready to deploy the search engine on idr

sbesson commented:
This is now ready to update and consume the production IDR

jburel commented:
@pwalczysko I have updated the url i.e. idr.o.org instead of idr-testing.o.rg and moved the variables around.
Could you give a final check to be sure I have not broken anything.

We can add more to this notebook in a second round

pwalczysko commented:
> Could you give a final check to be sure I have not broken anything.

Worked fine on MyBinder.

jburel commented:
thanks
